File: docs/tutorial/magics/projections/solution/multiprojection.py
# importing Magics module
from Magics.macro import *

# Definitions of the projection using a dictionary to 
# organise the informations
cylindrical = { "project" : "global",
                "text" : "Global Cylindrical",
                "map" : mmap({
                  "subpage_upper_right_longitude": 180.,
                  "subpage_upper_right_latitude": 90.,
                  "subpage_lower_left_longitude": -180.,
                  "subpage_map_projection": "cylindrical",
                  "subpage_lower_left_latitude": -90.})
              }

# Add the other areas here 
europe = { "project" : "europe",
           "text" : "Europe",
           "map" : mmap({
                  "subpage_lower_left_latitude": 21.51,
                  "subpage_lower_left_longitude": -37.27,
                  "subpage_upper_right_latitude": 51.28, 
                  "subpage_upper_right_longitude": 65., 
                  "subpage_map_projection": "polar_stereographic" })
         }

north_america = { "project" : "North America",
                  "text" : "North America",
                  "map" : mmap({
                    "subpage_map_vertical_longitude": -100., 
                    "subpage_lower_left_latitude": -5.,
                    "subpage_lower_left_longitude": -140.,
                    "subpage_upper_right_latitude": 30.,
                    "subpage_upper_right_longitude": -15.,
                    "subpage_map_projection": "polar_stereographic" })
                }

south_america = { "project" : "South America",
                  "text" : "South America",
                  "map" : mmap({
                    "subpage_map_vertical_longitude": 0., 
                    "subpage_lower_left_latitude": -65.,
                    "subpage_lower_left_longitude": -125.,
                    "subpage_upper_right_latitude": 20., 
                    "subpage_upper_right_longitude": 5., 
                    "subpage_map_projection": "cylindrical" })
                }

asia = { "project" : "Asia",
         "text" : "Asia",
         "map" : mmap({
                    "subpage_map_vertical_longitude": 0., 
                    "subpage_lower_left_latitude": 0.,
                    "subpage_lower_left_longitude": 55.,
                    "subpage_upper_right_latitude": 80., 
                    "subpage_upper_right_longitude": 175., 
                    "subpage_map_projection": "cylindrical" })
       }

australia = { "project" : "Australia",
              "text" : "Australia",
              "map" : mmap({
                    "subpage_map_vertical_longitude": 0., 
                    "subpage_lower_left_latitude": -55.,
                    "subpage_lower_left_longitude": 80.,
                    "subpage_upper_right_latitude": -5., 
                    "subpage_upper_right_longitude": 190., 
                    "subpage_map_projection": "cylindrical" })
            }

africa = { "project" : "Africa",
           "text" : "Africa",
           "map" : mmap({
                    "subpage_map_vertical_longitude": 0., 
                    "subpage_lower_left_latitude": -40.,
                    "subpage_lower_left_longitude": -45.,
                    "subpage_upper_right_latitude": 40., 
                    "subpage_upper_right_longitude": 75., 
                    "subpage_map_projection": "cylindrical" })
         }

# Create the list of the areas 
projections=[cylindrical, europe, north_america, south_america, asia, australia, africa]

# Create a empty list to store the objects to plot!
toplot = []

# Setting of the output file name
output = output({'output_name':'multi_projection'})

# store the output in the list

toplot = toplot + [output]

# Loop on the projections
for  projection in projections:

	# create a new page
	newpage = page()

	print projection

	# Setting the coordinates of the geographical area
	map=projection["map"]


	# Setting the coastlines attributes
	coast = mcoast( {"map_coastline_sea_shade_colour": "white", 
			"map_coastline_land_shade_colour": "cream", 
			"map_grid": "on", 
			"map_grid_colour": "grey", 
			"map_coastline_land_shade": "on", 
			"map_coastline_sea_shade": "on", 
			"map_label": "off", 
			"map_coastline_colour": "grey"})

	#Setting of the title
	text = mtext({'text_lines': [projection["text"]]})

	#Add to the list of things to plot
	toplot = toplot + [newpage, map, coast, text]

# end of the loop

# Plot the result
plot(toplot)

File: docs/tutorial/magics/projections/solution/multiprojection.f
C (C) Copyright 1996-2016 ECMWF.
C 
C This software is licensed under the terms of the Apache Licence Version 2.0
C which can be obtained at http://www.apache.org/licenses/LICENSE-2.0. 
C In applying this licence, ECMWF does not waive the privileges and immunities 
C granted to it by virtue of its status as an intergovernmental organisation nor
C does it submit to any jurisdiction.
C

      program  magics

C open Magics
      call popen

C Define the name of the output 
      call psetc('output_name', 'multi_projection')


      call area('Global Cylindrical', 'cylindrical        ', 
     + -90., -180., 90., 180., 0., 'north')
	  call area('Europe            ', 'polar_stereographic', 
     + 21.51, -37.27, 51.28, 65., 0., 'north')
      call area('North America     ', 'polar_stereographic', 
     + -5.0, -140.0, 30.0, -15.0, -100., 'north')
      call area('South America     ', 'cylindrical        ', 
     + -65.0, -125.0, 20.0, 5.0, 0., 'north')
      call area('Asia              ', 'cylindrical        ', 
     + 0.00, 55.00, 80.00, 175.00, 0., 'north')
      call area('Australia         ', 'cylindrical        ', 
     + -55.00, 80.0, -5.0, 190.0, 0., 'north')
      call area('Africa            ', 'cylindrical        ', 
     + -40.0, -45.00, 40.00, 75.00, 0., 'north')

      call pclose

      end 
C The subroutine will take as parameterthe information 
C necessary to build up the projection 
	  subroutine area(name, projection, lllat, lllon, 
     +                urlat, urlon, vertlon, hemis)

	  character*18 name
	  character*19 projection
	  real lllon, lllat
	  real urlon, urlat
	  real vertlon
	  character*6 hemis

C Create a new page in the postscript
	  call pnew('page')
C define the coordinates of the geographical area 
      call psetc('subpage_map_projection', projection)
      call psetr('subpage_lower_left_longitude', lllon)
      call psetr('subpage_upper_right_longitude', urlon)
      call psetr('subpage_upper_right_latitude', urlat)
      call psetr('subpage_lower_left_latitude', lllat)
      call psetr('subpage_map_vertical_longitude', vertlon)

C define the coastlines attributes 
      call psetc('map_coastline_colour', 'grey')
      call psetc('map_coastline_land_shade', 'on')
      call psetc('map_coastline_sea_shade', 'on')
      call psetc('map_coastline_land_shade_colour', 'cream')
      call psetc('map_coastline_sea_shade_colour', 'white')
      call psetc('map_grid', 'on')
      call psetc('map_grid_colour', 'grey')
      call psetc('map_label', 'off')
	
C Call the action routine to plot the coastlines
      call pcoast

C define the text to plot 
      call psetc('text_line_1', name)
	  call pseti('text_line_count', 1)

C Alternative setting : using an array of string instead
c     call pset1c('text_lines', (/'Global Map!'/), 1)

      call ptext
      
	  end
